teamworkApp/lib/importData.py

```diff
@@ -1,26 +1,256 @@
+			answer_to_db += [
+			(response,
+			student_dict[student][0],
+			now,
+			primary_keys[count],
+			question_count)
+			]
+			question_count += 1
+		count += 1
+	return answer_to_db
+
+def list_style_data(student_dict):
+	"""Given a dictionary of student data, calculate and import style data
+	for each student.
+	Uses find_style to calculate styles."""
+	style_to_db = []
+	# fetch the names of the students from the table
 	conn = sqlite3.connect(DB)
 	c = conn.cursor()
+	# get the primary keys for the students
+	c.execute('SELECT id FROM students;')
+	temp_keys = c.fetchall()
+	conn.commit()
+	conn.close()
+
+	primary_keys = [temp_key[0] for temp_key in temp_keys]
 
-	c.execute('SELECT * FROM students')
+	count = 0
+	for student in student_dict.keys():
+		now               = datetime.now().isoformat()
+		expanded_responses = student_dict[student][1]
+		communicator, collaborator, challenger, contributor = find_style(expanded_responses)
+		# TODO: The commented out line below should be what we use in the
+			# future; however, evaluateAnswers has not been updated for the
+			# new schema (as of the time of writing this 10-28-2017) so we
+			# cannot use it yet
+		# (communicator, collaborator, challenger, contributor) = find_scores(primary_keys[count])
+		created_at   = student_dict[student][0]
+		student_id   = primary_keys[count]
+		style_to_db += [
+			(student_id,
+			communicator,
+			collaborator,
+			challenger,
+			contributor,
+			created_at,
+			now)
+			]
+		count += 1
 
-	print(c.fetchone())
+	return style_to_db
+
+def execute_insert(csv_filename):
+	student_dict   = process_answer_data(csv_filename)
+	student_to_db = liststudent_data(student_dict)
+	print("Inserting .csv information into 'students'...")
+	conn = sqlite3.connect(DB)
+	c = conn.cursor()
+	# insert student data from .csv into students
+	c.executemany('INSERT INTO students (name, created_at, updated_at, username) VALUES (?, ?, ?, ?);', student_to_db)
+	conn.commit()
+	conn.close()
+	print("Success.")
+	# answer and style depend on information being stored in students so we
+		# can only call list_answer_data and liststudent_data after
+		# liststudent_data's information has been inserted
+	answer_to_db  = list_answer_data(student_dict)
+	style_to_db   = list_style_data(student_dict)
+	print("Inserting data into 'answers' and 'styles'...")
 
+	conn = sqlite3.connect(DB)
+	c = conn.cursor()
+	# insert answer data for each question for the student
+	c.executemany('INSERT INTO answers (value, created_at, updated_at, student_id, question) VALUES (?, ?, ?, ?, ?);', answer_to_db)
+	# insert style data for each student
+	c.executemany('INSERT INTO styles (student_id, communicator, collaborator, challenger, contributor, created_at, updated_at) VALUES (?, ?, ?, ?, ?, ?, ?);', style_to_db)
 	conn.commit()
 	conn.close()
 
+	print("Success.")
+
+def main():
+
+	print("Inserting data")
+	execute_insert('lib/test_spreadsheet.csv')
+	print("Success: Data inserted into 3 tables!")
+
```

Comment:
I like the print statements for now, I'm just worried about how they will react when we call this from ruby.

Reply:
Agreed. Once we get this up on Ruby and we know it works I'll delete them. I'll add it as a todo